chain/ethereum/src/adapter.rs

```diff
@@ -158,8 +158,10 @@ impl bc::TriggerFilter<Chain> for TriggerFilter {
 
     fn to_firehose_filter(self) -> Vec<prost_types::Any> {
         let EthereumBlockFilter {
+            polling_intervals: _polling_intervals,
```

Comment:
@leoyvens can something be done here for `to_firehose_filter`, can we optimize the firehose filters based on polling intervals if `trigger_on_every_block` is false and only certain blocks are required?

Here it seems `send_all_block_headers` is only true when trigger_on_every_block is true. Does that mean this feature wont work with firehose?

```
 let combined_filter = CombinedFilter {
            log_filters,
            call_filters,
            send_all_block_headers: trigger_every_block,
        };
 ```

Reply:
Thanks for raising attention this point, I don't think we could optimize it without adding support for this on the Firehose side.

It does look like the current code would have a bug when syncing with Firehose that would result in missed block triggers. If any `polling_intervals` exist, then we should set `send_all_block_headers` in the Firehose filter.

Reply:
Only possibility I could see would be to use the `HeaderOnly` filter, if you only need the `BlockHeader`, it's a good idea to use to avoid the network transfer of many useless bytes.

But like Leo said, Firehose would need to be adapted to send block each X block.

Reply:
I've modified this to set `send_all_block_headers` true when `polling_intervals` are not empty